rpcs3/Emu/Cell/Modules/cellSpursSpu.cpp

```diff
@@ -120,24 +121,30 @@ void cellSpursModuleExit(spu_thread& spu)
 }
 
 // Execute a DMA operation
-bool spursDma(spu_thread& spu, u32 cmd, u64 ea, u32 lsa, u32 size, u32 tag)
+bool spursDma(spu_thread& spu, const spu_mfc_cmd& args)
 {
-	spu.set_ch_value(MFC_LSA, lsa);
-	spu.set_ch_value(MFC_EAH, static_cast<u32>(ea >> 32));
-	spu.set_ch_value(MFC_EAL, static_cast<u32>(ea));
-	spu.set_ch_value(MFC_Size, size);
-	spu.set_ch_value(MFC_TagID, tag);
-	spu.set_ch_value(MFC_Cmd, cmd);
-
-	if (cmd == MFC_GETLLAR_CMD || cmd == MFC_PUTLLC_CMD || cmd == MFC_PUTLLUC_CMD)
+	spu.ch_mfc_cmd = args;
+
+	if (!spu.process_mfc_cmd())
+	{
+		spu_runtime::g_escape(&spu);
+	}
+
+	if (args.cmd == MFC_GETLLAR_CMD || args.cmd == MFC_PUTLLC_CMD || args.cmd == MFC_PUTLLUC_CMD)
```

Comment:
This construct looks to just be:
if(arg.cmd==MFC_PUTLLC_CMD)
  return static_cast<u32>(spu.get_ch_value(MFC_RdAtomicStat))==0;
else
  return true;

Am I missing something?

Reply:
I would like to keep RdAtomicStat empty, its a horrible hack to keep it non-empty after this command.

Reply:
Simplifoied the code though.